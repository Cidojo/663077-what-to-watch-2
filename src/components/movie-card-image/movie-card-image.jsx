import * as React from 'react';
import PropTypes from 'prop-types';

const MovieCardImage = (props) => {
  return (
    <div className="small-movie-card__image">
      {props.renderVideo()}
    </div>
  );
};

MovieCardImage.propTypes = {
  renderVideo: PropTypes.func.isRequired
};

MovieCardImage.defaultProps = {
  renderVideo: () => {}
};

export {MovieCardImage};
