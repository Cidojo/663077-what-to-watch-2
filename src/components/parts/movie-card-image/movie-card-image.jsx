import * as React from 'react';
import * as PropTypes from 'prop-types';

import Player from './../player/player.jsx';
import withAutoplay from './../../../hocs/with-autoplay/with-autoplay.jsx';

import {SMALL_MOVIE_CARD_MOUSE_ENTER_DELAY} from '../../../constants/constants.js';
import {movieCardPropTypes} from "../../../global-custom-types";

const MovieCardImage = (props) => {
  let timerId = null;

  const {
    card,
    autoplay,
    onAutoplayStatusUpdate
  } = props;

  const _handleMovieCardMouseEnter = () => {
    timerId = setTimeout(() => onAutoplayStatusUpdate(true), SMALL_MOVIE_CARD_MOUSE_ENTER_DELAY);
  };

  const _handleMovieCardMouseLeave = () => {
    clearTimeout(timerId);

    onAutoplayStatusUpdate(false);
  };

  return (
    <div
      className="small-movie-card__image"
      onMouseEnter={_handleMovieCardMouseEnter}
      onMouseLeave={_handleMovieCardMouseLeave}
    >
      <Player
        card={card}
        autoplay={autoplay}
        controls={false}
        muted={true}
      />
    </div>
  );
};

MovieCardImage.propTypes = {
  card: movieCardPropTypes,
  autoplay: PropTypes.bool,
  onAutoplayStatusUpdate: PropTypes.func
};

MovieCardImage.defaultProps = {
  card: {},
  autoplay: false,
  onAutoplayStatusUpdate: () => {}
};

export {MovieCardImage};
export default withAutoplay(MovieCardImage);
