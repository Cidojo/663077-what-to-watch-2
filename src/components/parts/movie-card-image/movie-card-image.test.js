import * as React from 'react';
import {MovieCardImage} from './movie-card-image.jsx';
import * as renderer from 'react-test-renderer';

const props = {
  card: {},
  autoplay: false,
  onAutoplayStatusUpdate: () => {}
};

it(`should render MovieCardImage component without errors`, () => {
  const tree = renderer.create(<MovieCardImage {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
