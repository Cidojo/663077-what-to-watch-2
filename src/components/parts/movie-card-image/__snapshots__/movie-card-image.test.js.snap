// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieCardImage component without errors 1`] = `
<div
  className="small-movie-card__image"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <video
    className="player__video"
    muted={true}
    poster=""
    src=""
  />
</div>
`;
